src/sidebar/cross-origin-rpc.js

```diff
@@ -93,8 +107,34 @@ function start(store, settings, $window) {
   }
 }
 
+/**
+ * Queues all incoming RPC requests so they can be handled later.
+ *
+ * @param {object} event - RPC event payload
+ */
+function preStartMessageListener(event) {
+  preStartQueue.push(event);
+}
+
+/**
+ * Start listening to incoming RCP requests right away so they don't timeout. These
+ * requests are saved until the server starts up and then they can be handled accordingly.
+ *
+ * Why?
+ *
+ * This allows the client to fetch its config from the parent app frame and potentially
+ * receive new unsolicited incoming requests that the parent app may send before this
+ * RCP server starts.
+ *
+ * @param {Window} $window - Test seam
+ */
+function preStart($window = window) {
+  $window.addEventListener('message', preStartMessageListener);
+}
+
 export default {
```

Comment:
It looks like I didn't adjust the export style to be idiomatic ES modules exports properly here. I think it would make more sense for this module to export `startServer` and `preStartServer` functions directly rather than default-exporting some object.